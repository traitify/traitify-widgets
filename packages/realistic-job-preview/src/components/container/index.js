import PropTypes from "prop-types";
import Context from "lib/context";
import ErrorBoundary from "./error-boundary";
import Theme from "./theme";
import useReducer from "./use-reducer";

function Container({children, className = null, ...props}) {
  const state = useReducer({...props});

  return (
    <Context.Provider value={state}>
      <Theme className={className}>
        <ErrorBoundary>{children}</ErrorBoundary>
      </Theme>
    </Context.Provider>
  );
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};

export default Container;
