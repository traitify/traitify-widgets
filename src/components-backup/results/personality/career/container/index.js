import PropTypes from "prop-types";
import TraitifyPropTypes from "lib/helpers/prop-types";
import useDidMount from "lib/hooks/use-did-mount";
import useDidUpdate from "lib/hooks/use-did-update";
import withTraitify from "lib/with-traitify";
import CareerFilter from "../filter";
import CareerList from "../list";
import CareerModal from "../modal";
import style from "./style.scss";

function CareerContainer(props) {
  const {ui, isReady, setElement} = props;
  const state = {};

  useDidMount(() => { ui.trigger("Careers.initialized", {props, state}); });
  useDidUpdate(() => { ui.trigger("Careers.updated", {props, state}); });

  return (
    isReady("results") && (
      <div className={style.container} ref={setElement}>
        <CareerFilter {...props} />
        <CareerList {...props} />
        <CareerModal {...props} />
      </div>
    )
  );
}
CareerContainer.propTypes = {
  isReady: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired,
  setElement: PropTypes.func.isRequired
};

export {CareerContainer as Component};
export default withTraitify(CareerContainer);
