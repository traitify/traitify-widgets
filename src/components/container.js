/* eslint-disable react/forbid-prop-types */
import PropTypes from "prop-types";
import {useEffect} from "react";
import {RecoilRoot, useRecoilState, useSetRecoilState} from "recoil";
import slice from "lib/common/object/slice";
import Http from "lib/http";
import I18n from "lib/i18n";
import Listener from "lib/listener";
import {
  assessmentIDState,
  benchmarkIDState,
  httpState,
  i18nState,
  listenerState,
  loadingState,
  localeState,
  optionsState,
  profileIDState
} from "lib/recoil";

function State({children, http: _http, i18n: _i18n, listener: _listener, options}) {
  const setAssessmentID = useSetRecoilState(assessmentIDState);
  const setBenchmarkID = useSetRecoilState(benchmarkIDState);
  const setHttp = useSetRecoilState(httpState);
  const setI18n = useSetRecoilState(i18nState);
  const setListener = useSetRecoilState(listenerState);
  const setLocale = useSetRecoilState(localeState);
  const setOptions = useSetRecoilState(optionsState);
  const setProfileID = useSetRecoilState(profileIDState);
  const [loading, setLoading] = useRecoilState(loadingState);
  const {
    assessmentID,
    benchmarkID,
    locale,
    profileID,
    ...extraOptions
  } = options;

  useEffect(() => {
    setOptions(extraOptions);
  }, [options]);

  useEffect(() => {
    if(!locale) { return; }

    setLocale(locale);
  }, [locale]);

  useEffect(() => {
    const http = _http || new Http(slice(options, ["authKey", "host", "version"]));

    setHttp(http);
  }, [_http]);

  useEffect(() => {
    const i18n = _i18n || new I18n();

    setI18n(i18n);
  }, [_i18n]);

  useEffect(() => {
    const listener = _listener || new Listener();

    setListener(listener);
  }, [_listener]);

  useEffect(() => {
    setAssessmentID(assessmentID);
  }, [assessmentID]);

  useEffect(() => {
    setBenchmarkID(benchmarkID);
  }, [benchmarkID]);

  useEffect(() => {
    setProfileID(profileID);
  }, [profileID]);

  useEffect(() => {
    setLoading(false);
  }, []);

  if(loading) { return null; }

  return children;
}

State.defaultProps = {
  http: null,
  i18n: null,
  listener: null,
  options: {}
};
State.propTypes = {
  children: PropTypes.node.isRequired,
  http: PropTypes.object,
  i18n: PropTypes.object,
  listener: PropTypes.object,
  options: PropTypes.shape({
    assessmentID: PropTypes.string,
    benchmarkID: PropTypes.string,
    locale: PropTypes.string,
    profileID: PropTypes.string
  })
};

export default function Container(props) {
  return (
    <RecoilRoot>
      <State {...props} />
    </RecoilRoot>
  );
}
