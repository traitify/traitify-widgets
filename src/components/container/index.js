import PropTypes from "prop-types";
import {RecoilRoot} from "recoil";
import ErrorBoundary from "./error-boundary";
import State from "./state";
import Theme from "./theme";

function Container(props) {
  return (
    <RecoilRoot>
      <Theme className={props.className}>
        <ErrorBoundary>
          <State {...props} />
        </ErrorBoundary>
      </Theme>
    </RecoilRoot>
  );
}

Container.defaultProps = {className: null};
Container.propTypes = {className: PropTypes.string};

export default Container;
