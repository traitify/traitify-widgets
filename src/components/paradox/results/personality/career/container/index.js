import PropTypes from "prop-types";
import {useEffect} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import CareerFilter from "../filter";
import CareerModal from "../modal";
import CareerResults from "../list";
import style from "./style.scss";

function Careers(props) {
  const {ui, isReady} = props;

  useEffect(() => {
    ui.trigger("Careers.initialized", {props});
  }, []);
  useEffect(() => {
    ui.trigger("Careers.updated", {props});
  });

  return (
    isReady("results") && (
      <div className={style.container}>
        <CareerFilter {...props} />
        <CareerResults {...props} />
        <CareerModal {...props} />
      </div>
    )
  );
}
Careers.propTypes = {
  isReady: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};
export {Careers as Component};
export default withTraitify(Careers);
