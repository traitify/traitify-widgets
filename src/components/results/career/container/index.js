import useComponentEvents from "lib/hooks/use-component-events";
import useResults from "lib/hooks/use-results";
import CareerFilter from "../filter";
import CareerList from "../list";
import CareerModal from "../modal";
import style from "./style.scss";

export default function CareerContainer() {
  const results = useResults();

  useComponentEvents("Careers");

  if(!results) { return null; }

  return (
    <div className={style.container}>
      <CareerFilter />
      <CareerList />
      <CareerModal />
    </div>
  );
}
