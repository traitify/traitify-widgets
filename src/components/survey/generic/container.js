import PropTypes from "prop-types";
import style from "./style.scss";

export default function Container({children, progress}) {
  return (
    <div className={`${style.container}`}>
      {progress < 100
        && (
        <div className={style.progressBar}>
          <div className={style.progress} style={{width: `${progress}%`}} />
        </div>
        )}
      {children}
    </div>
  );
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  progress: PropTypes.number.isRequired
};
