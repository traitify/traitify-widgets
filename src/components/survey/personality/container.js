import {
  faChevronLeft,
  faMaximize,
  faMinimize
} from "@fortawesome/free-solid-svg-icons";
import PropTypes from "prop-types";
import {useEffect, useLayoutEffect, useRef} from "react";
import Icon from "components/common/icon";
import dig from "lib/common/object/dig";
import useAssessment from "lib/hooks/use-assessment";
import useElementSize from "lib/hooks/use-element-size";
import useFullscreen from "lib/hooks/use-fullscreen";
import useOption from "lib/hooks/use-option";
import useTranslate from "lib/hooks/use-translate";
import Responses from "./responses";
import style from "./style.scss";

function Container({
  caption = null,
  children,
  dispatch,
  likert = false,
  onResponse = null,
  progress,
  slideIndex
}) {
  const assessment = useAssessment({surveyType: "personality"});
  const {allowFullscreen, ...options} = useOption("survey") || {};
  const container = useRef(null);
  const content = useRef(null);
  const [fullscreen, toggleFullscreen] = useFullscreen(container.current);
  const size = useElementSize(content.current);
  const text = useRef(null);
  const translate = useTranslate();

  useEffect(() => { dispatch({size, type: "resize"}); }, [...size]);
  useLayoutEffect(() => {
    if(!text.current) { return; }

    text.current.focus({preventScroll: true});
  }, [caption]);

  const back = () => dispatch({type: "back"});
  const textSurvey = dig(assessment, "slide_type")?.toLowerCase() === "text";
  const allowBack = Object.hasOwn(options, "allowBack") ? options.allowBack : textSurvey;

  return (
    <div className={`${style.container} traitify--survey-${textSurvey ? "text" : "image"}`} ref={container}>
      {caption && !textSurvey && (
        <div className={style.caption} ref={text} tabIndex="-1">{caption}</div>
      )}
      {textSurvey && (
        <div className={style.progressBar}>
          <div className={style.progress} style={{width: `${progress}%`}} />
        </div>
      )}
      <div className={style.content} ref={content}>
        {!textSurvey && <div className={style.progress} style={{width: `${progress}%`}} />}
        {children}
        {allowBack && slideIndex > 0 && (
          <button className={style.back} onClick={back} type="button">
            <Icon alt={translate("back")} icon={faChevronLeft} />
          </button>
        )}
        {allowFullscreen && (
          <button className={style.fullscreen} onClick={toggleFullscreen} type="button">
            <Icon alt="fullscreen" icon={fullscreen ? faMinimize : faMaximize} />
          </button>
        )}
      </div>
      <Responses likert={likert} onResponse={onResponse} />
    </div>
  );
}

Container.propTypes = {
  caption: PropTypes.string,
  children: PropTypes.node.isRequired,
  dispatch: PropTypes.func.isRequired,
  likert: PropTypes.bool,
  onResponse: PropTypes.func,
  progress: PropTypes.number.isRequired,
  slideIndex: PropTypes.number.isRequired
};

export default Container;
