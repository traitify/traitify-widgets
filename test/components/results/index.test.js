import Component from "components/results";
import ComponentHandler from "support/component-handler";
import {mockAssessment} from "support/container/http";
import {mockOption} from "support/container/options";
import useContainer from "support/hooks/use-container";
import dimensionBased from "support/json/assessment/dimension-based.json";
import financialRisk from "support/json/assessment/financial-risk.json";
import typeBased from "support/json/assessment/type-based.json";

jest.mock("components/report/attract", () => (() => <div className="mock">Attract</div>));
jest.mock("components/report/candidate", () => (() => <div className="mock">Candidate</div>));
jest.mock("components/report/employee", () => (() => <div className="mock">Employee</div>));
jest.mock("components/report/manager", () => (() => <div className="mock">Manager</div>));
jest.mock("components/results/cognitive", () => (() => <div className="mock">Cognitive</div>));
jest.mock("components/results/financial-risk", () => (() => <div className="mock">Financial Risk</div>));

describe("Results", () => {
  let component;

  useContainer();

  describe("callbacks", () => {
    it("triggers initialization", async() => {
      await ComponentHandler.setup(Component);

      expect(container.listener.trigger).toHaveBeenCalledWith(
        "Results.initialized",
        undefined
      );
    });

    it("triggers update", async() => {
      component = await ComponentHandler.setup(Component);
      await component.update();

      expect(container.listener.trigger).toHaveBeenCalledWith(
        "Results.updated",
        undefined
      );
    });
  });

  it("renders candidate report", async() => {
    mockAssessment(dimensionBased);
    mockOption("report", "candidate");
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders cognitive results", async() => {
    mockOption("surveyType", "cognitive");
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders dimension based results", async() => {
    mockAssessment(dimensionBased);
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders employee report", async() => {
    mockAssessment(dimensionBased);
    mockOption("report", "employee");
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders financial risk results", async() => {
    mockAssessment(financialRisk);
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders manager report", async() => {
    mockAssessment(dimensionBased);
    mockOption("report", "employee");
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders attract report", async() => {
    mockAssessment(typeBased);
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if not ready", async() => {
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });
});
