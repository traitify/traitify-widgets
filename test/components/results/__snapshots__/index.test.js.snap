// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results renders attract report 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
>
  <div
    className="mock"
  >
    Attract
  </div>
</div>
`;

exports[`Results renders candidate report 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
>
  <div
    className="mock"
  >
    Candidate
  </div>
</div>
`;

exports[`Results renders cognitive results 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
>
  <div
    className="mock"
  >
    Cognitive
  </div>
</div>
`;

exports[`Results renders dimension based results 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
>
  <div
    className="mock"
  >
    Candidate
  </div>
</div>
`;

exports[`Results renders employee report 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
>
  <div
    className="mock"
  >
    Employee
  </div>
</div>
`;

exports[`Results renders financial risk results 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
>
  <div
    className="mock"
  >
    Financial Risk
  </div>
</div>
`;

exports[`Results renders manager report 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
>
  <div
    className="mock"
  >
    Manager
  </div>
</div>
`;

exports[`Results renders nothing if not ready 1`] = `
<div
  className="traitify--container traitify--color-scheme-light"
/>
`;
